sqlglot/dialects/dialect.py

```diff
@@ -17,6 +17,15 @@
 B = t.TypeVar("B", bound=exp.Binary)
 
 
+def _cast_div_operands_sql(self: Generator, expression: exp.Div) -> str:
+    l, r = expression.left, expression.right
+
+    if not l.is_type(*exp.DataType.FLOAT_TYPES) and not r.is_type(*exp.DataType.FLOAT_TYPES):
+        l.replace(exp.cast(l.copy(), to=exp.DataType.Type.DOUBLE))
```

Comment:
Should this happen conditionally, only when writing towards dialects that support float div semantics?

Reply:
Not sure I follow. But this _is_ being used conditionally

Reply:
Ah we set it at the metaclass and I think when I made the comment I hadn't noticed yet.

So, coming back to (what I think was) Toby's comment, should we do this @ the method level and not at the metaclass, like:

```python
if self.TYPED_DIVISION and (above logic):
    l.replace(exp.cast(l.copy(), to=exp.DataType.Type.DOUBLE))
```